src/vs/editor/browser/view/viewController.ts

```diff
@@ -133,8 +133,11 @@ export class ViewController {
 		const options = this.configuration.options;
 		const selectionClipboardIsOn = (platform.isLinux && options.get(EditorOption.selectionClipboard));
 		const columnSelection = options.get(EditorOption.columnSelection);
-		if (data.middleButton && !selectionClipboardIsOn) {
-			this._columnSelect(data.position, data.mouseColumn, data.inSelectionMode);
+		const scrollOnMiddleClick = options.get(EditorOption.scrollOnMiddleClick);
+		if (data.middleButton && !selectionClipboardIsOn && (columnSelection || scrollOnMiddleClick)) {
+			if (columnSelection) {
```

Comment:
This fixes column select on middle button when editor.columnSelection setting is off